drivers/i2c/gpio_i2c_switch.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Ayush Singh <ayushdevel1325@gmail.com>
+ * Copyright (c) 2021 Jason Kridner, BeagleBoard.org Foundation
+ * Copyright (c) 2020 Innoseis BV
+ *
+ * Based on i2c_tca9656a.c
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_i2c_switch
+
+#define GPIO_I2C_TRANSFER_DELAY_US 100
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(gpio_i2c_switch, CONFIG_I2C_LOG_LEVEL);
+
+struct gpio_i2c_switch_config {
+	const struct device *bus;
+	const struct gpio_dt_spec gpio;
+};
+
+struct gpio_i2c_switch_data {
+	struct k_mutex lock;
+};
+
+static int gpio_i2c_switch_configure(const struct device *dev, uint32_t dev_config)
+{
+	const struct gpio_i2c_switch_config *config = dev->config;
+
+	return i2c_configure(config->bus, dev_config);
+}
+
+static int gpio_i2c_switch_enable(const struct device *dev)
+{
+	const struct gpio_i2c_switch_config *config = dev->config;
+
+	return gpio_pin_configure(config->gpio.port, config->gpio.pin, GPIO_OUTPUT_HIGH);
```

Comment:
`gpio_pin_configure_dt`, to be honest at this point you may as well get rid of the function and call this from `gpio_i2c_switch_transfer` directly